cilium 1.18 image fails to build on rootless buildkit

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Version

equal or higher than v1.18.1 and lower than v1.19.0

### What happened?

The [cilium image](https://github.com/cilium/cilium/tree/3f0383ba60fa639401576af1f62100114ae7832a/images/cilium) fails to build on [rootless buildkit](https://github.com/moby/buildkit/blob/master/docs/rootless.md) since cilium 1.18.0.

#### Why this happens:

The cilium image build generates bash completion scripts for the `cilium-dbg` CLI [here](https://github.com/cilium/cilium/blob/3f0383ba60fa639401576af1f62100114ae7832a/images/cilium/Dockerfile#L54) by running the [`install-bash-completion`](https://github.com/cilium/cilium/blob/3f0383ba60fa639401576af1f62100114ae7832a/Makefile#L210) make target which itself calls the [same target](https://github.com/cilium/cilium/blob/3f0383ba60fa639401576af1f62100114ae7832a/cilium-dbg/Makefile#L37-L42) in cilium-dbg which executes the [`cilium-dbg completion bash`](https://github.com/cilium/cilium/blob/3f0383ba60fa639401576af1f62100114ae7832a/cilium-dbg/Makefile#L41) command.

Since #37591, running `cilium-dbg` leads to evaluating [`ciliumebpf.MustPossibleCPU()`](https://github.com/cilium/ebpf/blob/ae226118949d4e3de64520195b66a09591116ea0/cpu.go#L11) through the `pkg/maps/policymap` package global variable [`nCPU`](https://github.com/cilium/cilium/blob/3f0383ba60fa639401576af1f62100114ae7832a/pkg/maps/policymap/statsmap.go#L31). The problem is that this function panics if it can't read `/sys/devices/system/cpu/possible` and when running buildkit in rootless mode, `/sys` [is not mounted](https://github.com/moby/buildkit/blob/f9b9b363d6a6777259e435987a1df0d6a314ec8a/util/rootless/specconv/specconv_linux.go#L15-L32) in the build container. This causes the `cilium-dbg completion bash` command to panic and fail the build.


### How can we reproduce the issue?

`docker buildx build -t cilium -f images/cilium/Dockerfile --target release .` with buildx configured to use a rootless buildkit backend

### Cilium Version

1.18.0+

### Kernel Version

-

### Kubernetes Version

-

### Regression

Versions < 1.18.0

### Sysdump

-

### Relevant log output

This is the failed docker build log 

```shell
------
 > [linux/amd64 builder 4/7] RUN --mount=type=bind,readwrite,target=/go/src/github.com/cilium/cilium     --mount=type=cache,target=/root/.cache     --mount=type=cache,target=/go/pkg     make GOARCH=${BUILDARCH} DESTDIR=/tmp/install/linux/amd64 PKG_BUILD=1 $(echo NOSTRIP=1 | tr -d '"')     install-bash-completion licenses-all &&     mv LICENSE.all /tmp/install/linux/amd64/LICENSE.all &&     mkdir -p /tmp/hubble/linux/amd64 &&     cd hubble &&     make GOOS=linux GOARCH=amd64 $(echo NOSTRIP=1 | tr -d '"') &&     mv hubble /tmp/hubble/linux/amd64/hubble:
4.941 panic: open /sys/devices/system/cpu/possible: no such file or directory
4.941 
4.941 goroutine 1 [running]:
4.941 github.com/cilium/ebpf.MustPossibleCPU()
4.941 	/go/src/github.com/cilium/cilium/vendor/github.com/cilium/ebpf/cpu.go:14 +0x37
4.941 github.com/cilium/cilium/pkg/maps/policymap.init()
4.941 	/go/src/github.com/cilium/cilium/pkg/maps/policymap/statsmap.go:31 +0x17
4.943 make[1]: Leaving directory '/go/src/github.com/cilium/cilium/cilium-dbg'
4.943 make[1]: *** [Makefile:41: install-bash-completion-only] Error 2
4.944 make: *** [Makefile:212: install-bash-completion] Error 2
------
Dockerfile:45
--------------------
  44 |     
  45 | >>> RUN --mount=type=bind,readwrite,target=/go/src/github.com/cilium/cilium \
  46 | >>>     --mount=type=cache,target=/root/.cache \
  47 | >>>     --mount=type=cache,target=/go/pkg \
  48 | >>>     # install-bash-completion will execute the bash_completion script. It is
  49 | >>>     # fine to run this with same architecture as BUILDARCH since the output of
  50 | >>>     # bash_completion is the same for both architectures.
  51 | >>>     make GOARCH=${BUILDARCH} DESTDIR=/tmp/install/${TARGETOS}/${TARGETARCH} PKG_BUILD=1 $(echo $MODIFIERS | tr -d '"') \
  52 | >>>     install-bash-completion licenses-all && \
  53 | >>>     mv LICENSE.all /tmp/install/${TARGETOS}/${TARGETARCH}/LICENSE.all && \
  54 | >>>     mkdir -p /tmp/hubble/${TARGETOS}/${TARGETARCH} && \
  55 | >>>     cd hubble && \
  56 | >>>     make GOOS=${TARGETOS} GOARCH=${TARGETARCH} $(echo $MODIFIERS | tr -d '"') && \
  57 | >>>     mv hubble /tmp/hubble/${TARGETOS}/${TARGETARCH}/hubble
  58 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c make GOARCH=${BUILDARCH} DESTDIR=/tmp/install/${TARGETOS}/${TARGETARCH} PKG_BUILD=1 $(echo $MODIFIERS | tr -d '\"')     install-bash-completion licenses-all &&     mv LICENSE.all /tmp/install/${TARGETOS}/${TARGETARCH}/LICENSE.all &&     mkdir -p /tmp/hubble/${TARGETOS}/${TARGETARCH} &&     cd hubble &&     make GOOS=${TARGETOS} GOARCH=${TARGETARCH} $(echo $MODIFIERS | tr -d '\"') &&     mv hubble /tmp/hubble/${TARGETOS}/${TARGETARCH}/hubble" did not complete successfully: exit code: 2
```

### Anything else?

_No response_

### Cilium Users Document

- [x] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/pkg/maps/policymap/statsmap.go
+++ b/pkg/maps/policymap/statsmap.go
@@ -26,11 +26,6 @@ const (
 	StatNotAvailable = uint64(math.MaxUint64)
 )
 
-var (
-	// nCPU must be on package level to be available for StatsValues.New() below
-	nCPU = ciliumebpf.MustPossibleCPU()
-)
-
 // PolicyStatsMap maps endpoint IDs to the fd for the program which
 // implements its policy.
 type StatsMap struct {
@@ -39,6 +34,7 @@ type StatsMap struct {
 }
 
 func newStatsMap(maxStatsEntries int, log *slog.Logger) (*StatsMap, int) {
+	nCPU := possibleCPU(log)
 	roundDown := maxStatsEntries % nCPU
 	maxStatsEntries -= roundDown
 
@@ -68,6 +64,16 @@ func OpenStatsMap(logger *slog.Logger) (*StatsMap, error) {
 		Map: m,
 		log: logger,
 	}, nil
+}
+
+func possibleCPU(logger *slog.Logger) int {
+	nCPU, err := ciliumebpf.PossibleCPU()
+	if err != nil {
+		logger.Error("Failed to determine possible CPUs", logfields.Error, err)
+		// Fallback to 1 CPU if we cannot determine the number of CPUs
+		return 1
+	}
+	return nCPU
 }
 
 type StatsKey struct {
@@ -166,6 +172,7 @@ func (m *StatsMap) ZeroStat(epID uint16, k PolicyKey) error {
 		Nexthdr:          k.Nexthdr,
 		DestPortNetwork:  k.DestPortNetwork,
 	}
+	nCPU := possibleCPU(m.log)
 	zeroValue := make(StatsValues, nCPU)
 
 	err := m.Update(&statsKey, zeroValue, 0)

